crates/sui-types/src/temporary_store.rs

```diff
@@ -98,7 +99,7 @@ pub struct TemporaryStore<S> {
     // into written directly.
     written: BTreeMap<ObjectID, (SingleTxContext, Object, WriteKind)>, // Objects written
     /// Objects actively deleted.
-    deleted: BTreeMap<ObjectID, (SingleTxContext, SequenceNumber, DeleteKind)>,
+    deleted: BTreeMap<ObjectID, (SingleTxContext, SequenceNumber, DeleteKind, Object)>,
```

Comment:
we don't need to store the whole Object, which can be quite big, storing the type information should be enough